packages/message/src/types.ts

```diff
@@ -1,15 +1,29 @@
-export type MessageTemplate = {
-  cumulus_meta: object
+import { Message } from '@cumulus/types';
+
+export interface MessageTemplateCumulusMeta {
+  queueExecutionLimits: {
+    [queueUrl: string]: number
+  }
+}
+
+export interface MessageTemplate {
+  cumulus_meta: MessageTemplateCumulusMeta
   meta: object
-};
+}
 
-export type Workflow = {
+export interface Workflow {
   arn: string
   name: string
-};
+}
 
-export type QueueMessageMeta = {
+export interface QueueMessageMeta {
   workflow_name: string
   collection?: object
   provider?: object
-};
+}
+
+export interface CumulusQueueMessage {
```

Comment:
I'm not sure I understand the distinction between stuff we define here vs. stuff we define in `@cumulus/types/message.d.ts`. How/why is this different from the `CumulusMessage` defined in `types`?

Reply:
The short answer is that the `CumulusMessage` type in the `types` package is more generic and intended for use in many places throughout our code. For example, the contents of `meta` are not easy to known when you're generally handling a cumulus message throughout our code, so `CumulusMessage.meta` is typed as a generic `object`. But in this case, we're building a cumulus message for a queue and we do know what `meta` will contain, so we're explicitly defining that type.

This pattern is generally what we're trying to follow - keep the types in `types` more generic and add more explicitly override/typing in specific scenarios where you can know the message contents better.

Reply:
This seems to make sense.